### PR TITLE
Feature: distributed function alter actions

### PR DESCRIPTION
**Pull request by thanodnl**
DESCRIPTION: Implement ALTER FUNCTION ... action [...] for distributed functions

TODO
 - [ ] Test for changes being applied remotely
 - [ ] Add tests for SET actions

### CODE REVIEW

File: src/backend/distributed/commands/function.c [hunk 1]
```diff
@@ -157,3 +161,90 @@ EnsureSequentialModeForFunctionDDL(void)
...
+
+const ObjectAddress *
+AlterFunctionStmtObjectAddress(AlterFunctionStmt *stmt, bool missing_ok)
+{
+	Oid funcOid = InvalidOid;
+	ObjectAddress *address = NULL;
+
+	funcOid = LookupFuncWithArgs(OBJECT_FUNCTION, stmt->func, missing_ok);
+	address = palloc0(sizeof(ObjectAddress));
+	ObjectAddressSet(*address, ProcedureRelationId, funcOid);
```
Comment (marcocitus) [inline src/backend/distributed/commands/function.c]: These 3 lines seem to repeat in several PRs for different calling contexts. That should just be a utility function that we used in multiple places.

File: src/backend/distributed/commands/function.c [hunk 2]
```diff
@@ -157,3 +161,90 @@ EnsureSequentialModeForFunctionDDL(void)
...
+	commands = list_make3(DISABLE_DDL_PROPAGATION,
+						  (void *) sql,
+						  ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+const ObjectAddress *
+AlterFunctionStmtObjectAddress(AlterFunctionStmt *stmt, bool missing_ok)
```
Comment (marcocitus) [inline src/backend/distributed/commands/function.c]: Needs a function comment. We do not expect readers to open this file with full knowledge of the dependency system.

File: src/backend/distributed/commands/function.c (hunk 1 above)
Comment (thanodnl) [reply src/backend/distributed/commands/function.c]: This has been implemented as a side effect of the comment on `forboth`, will use this function throughout.

File: src/backend/distributed/commands/function.c (hunk 2 above)
Comment (thanodnl) [reply src/backend/distributed/commands/function.c]: Will go through and comment all functions

**Comment by thanodnl** [comment]
This PR is superseded by #3041 feedback has been addressed there.

**Pull request closed without merging**